web/src/copy_and_paste.js

```diff
@@ -481,6 +481,61 @@ export function paste_handler_converter(paste_html) {
         },
     });
 
+    // We override the original upstream implementation of this rule to turn any
+    // single line code blocks into inline markdown code. Everything else is the same.
+    turndownService.addRule("fencedCodeBlock", {
+        filter(node, options) {
+            return (
+                options.codeBlockStyle === "fenced" &&
+                node.nodeName === "PRE" &&
+                node.firstChild &&
+                node.firstChild.nodeName === "CODE"
+            );
+        },
+
+        replacement(_content, node, options) {
+            let code = node.firstChild.textContent;
+
+            // We convert single line code inside a code block to inline markdown code,
+            // and the code for this is taken from upstream's `code` rule.
+            if (!code.includes("\n")) {
+                if (!code) {
+                    return "";
+                }
+                code = code.replaceAll(/\r?\n|\r/g, " ");
+
+                const extraSpace = /^`|^ .*?[^ ].* $|`$/.test(code) ? " " : "";
+                let delimiter = "`";
+                const matches = code.match(/`+/gm) || [];
+                while (matches.includes(delimiter)) {
+                    delimiter = delimiter + "`";
+                }
```

Comment:
I don't understand what this is doing either.

Reply:
This is to choose the delimiter such that it's the shortest sequence of backticks that does not already appear in the code, as we need to have a unique delimiter. Have added a comment explaining the same!